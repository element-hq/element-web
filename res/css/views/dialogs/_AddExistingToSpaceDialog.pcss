/*
Copyright 2024 New Vector Ltd.
Copyright 2021 The Matrix.org Foundation C.I.C.

SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only
Please see LICENSE files in the repository root for full details.
*/

.mx_AddExistingToSpaceDialog_wrapper {
    .mx_Dialog {
        display: flex;
        flex-direction: column;
    }
}

.mx_AddExistingToSpace {
    .mx_SearchBox {
        /* To match the space around the title */
        margin: 0 0 15px 0;
        flex-grow: 0;
    }

    .mx_AddExistingToSpace_content {
        flex-grow: 1;
    }

    .mx_AddExistingToSpace_noResults {
        display: block;
        margin-top: 24px;
    }

    .mx_AddExistingToSpace_section {
        margin-right: 12px;

        // provides space for scrollbar so that checkbox and scrollbar do not collide

        &:not(:first-child) {
            margin-top: 24px;
        }

        > h3 {
            margin: 0;
            color: $secondary-content;
            font-size: $font-12px;
            font-weight: var(--cpd-font-weight-semibold);
            line-height: $font-15px;
        }

        .mx_AccessibleButton_kind_link {
            font-size: $font-12px;
            line-height: $font-15px;
            margin-top: 8px;
        }
    }

    .mx_AddExistingToSpace_footer {
        display: flex;
        margin-top: 20px;

        > span {
            flex-grow: 1;
            font-size: $font-12px;
            line-height: $font-15px;
            color: $secondary-content;

            .mx_ProgressBar {
                height: 8px;
                width: 100%;

                @mixin ProgressBarBorderRadius 8px;
            }

            .mx_AddExistingToSpaceDialog_progressText {
                margin-top: 8px;
                font-size: $font-15px;
                line-height: $font-24px;
                color: $primary-content;
            }

            > * {
                vertical-align: middle;
            }
        }

        .mx_AddExistingToSpace_error {
            padding-left: 12px;

            > img {
                align-self: center;
            }

            .mx_AddExistingToSpace_errorHeading {
                font-weight: var(--cpd-font-weight-semibold);
                font-size: $font-15px;
                line-height: $font-18px;
                color: $alert;
            }

            .mx_AddExistingToSpace_errorCaption {
                margin-top: 4px;
                font-size: $font-12px;
                line-height: $font-15px;
                color: $primary-content;
            }
        }

        .mx_AccessibleButton {
            display: inline-block;
            align-self: center;
        }

        .mx_AccessibleButton_kind_primary {
            padding: 8px 36px;
        }

        .mx_AddExistingToSpace_retryButton {
            margin-left: 12px;
            padding-left: 24px;
            position: relative;

            &::before {
                content: "";
                position: absolute;
                background-color: $primary-content;
                mask-repeat: no-repeat;
                mask-position: center;
                mask-size: contain;
                mask-image: url("$(res)/img/element-icons/retry.svg");
                width: 18px;
                height: 18px;
                left: 0;
            }
        }
    }
}

.mx_AddExistingToSpaceDialog {
    width: 480px;
    color: $primary-content;
    display: flex;
    flex-direction: column;
    flex-wrap: nowrap;
    min-height: 0;
    height: 80vh;

    .mx_AddExistingToSpace {
        display: contents;
    }
}

.mx_SubspaceSelector {
    display: flex;

    .mx_BaseAvatar {
        margin: auto 16px auto 5px;
    }

    > div {
        > h1 {
            font-weight: var(--cpd-font-weight-semibold);
            font-size: $font-18px;
            line-height: $font-22px;
            margin: 0;
        }
    }

    .mx_Dropdown_input {
        border: none;

        > .mx_Dropdown_option {
            padding-left: 0;
            flex: unset;
            height: unset;
            color: $secondary-content;
            font-size: $font-15px;
            line-height: $font-24px;

            .mx_BaseAvatar {
                display: none;
            }
        }

        .mx_Dropdown_menu {
            .mx_SubspaceSelector_dropdownOptionActive {
                color: $accent;
                padding-right: 32px;
                position: relative;

                &::before {
                    content: "";
                    width: 20px;
                    height: 20px;
                    top: 8px;
                    right: 0;
                    position: absolute;
                    mask-position: center;
                    mask-size: contain;
                    mask-repeat: no-repeat;
                    background-color: $accent;
                    mask-image: url("@vector-im/compound-design-tokens/icons/check.svg");
                }
            }
        }
    }

    .mx_SubspaceSelector_onlySpace {
        color: $secondary-content;
        font-size: $font-15px;
        line-height: $font-24px;
    }
}

.mx_AddExistingToSpace_entry {
    display: flex;
    margin-top: 12px;

    .mx_DecoratedRoomAvatar, /* we can't target .mx_BaseAvatar here as it'll break the decorated avatar styling */ {
        margin-right: 12px;
    }

    .mx_AddExistingToSpace_entry_name {
        font-size: $font-15px;
        line-height: 30px;
        flex-grow: 1;
        overflow: hidden;
        white-space: nowrap;
        text-overflow: ellipsis;
        margin-right: 12px;
    }

    .mx_Checkbox {
        align-items: center;
    }
}
