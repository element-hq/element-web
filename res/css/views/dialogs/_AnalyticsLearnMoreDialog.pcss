/*
Copyright 2024 New Vector Ltd.
Copyright 2021 The Matrix.org Foundation C.I.C.

SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only
Please see LICENSE files in the repository root for full details.
*/

.mx_AnalyticsLearnMoreDialog {
    max-width: 500px;
    .mx_AnalyticsLearnMore_image_holder {
        background-image: url("$(res)/img/element-shiny.svg");
        background-repeat: no-repeat;
        background-position: center top;
        height: 112px;
        padding: 20px 0px;
    }

    .mx_Dialog_content {
        margin-bottom: 0px;
    }

    .mx_AnalyticsLearnMore_copy {
        border-bottom: 1px solid $menu-border-color;
        padding-bottom: 20px;
        margin-bottom: 20px;
    }

    a {
        color: $accent;
        text-decoration: none;
    }

    .mx_AnalyticsLearnMore_bullets {
        padding-left: 0px;
    }

    .mx_AnalyticsLearnMore_bullets li {
        background: url("$(res)/img/tick-circle.svg") no-repeat;
        list-style-type: none;
        padding: 2px 0px 20px 32px;
        vertical-align: middle;
    }
}
