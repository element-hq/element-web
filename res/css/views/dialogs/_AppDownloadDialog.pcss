.mx_AppDownloadDialog {
    display: flex;
    flex-direction: column;
    gap: $spacing-32;
    color: $primary-content;

    &.mx_Dialog_fixedWidth {
        width: 640px;
    }

    .mx_AppDownloadDialog_desktop {
        display: flex;
        flex-direction: column;
        align-items: center;
        gap: $spacing-16;
    }

    .mx_AppDownloadDialog_mobile {
        display: flex;
        flex-direction: row;
        gap: $spacing-24;

        .mx_AppDownloadDialog_app {
            display: flex;
            flex-direction: column;
            flex-grow: 1;
            flex-basis: 50%;
            align-items: center;
            gap: $spacing-16;

            .mx_QRCode {
                /* intentionally hardcoded color to ensure the QR code is readable in any situation */
                background: #ffffff;

                padding: $spacing-24;
                border: 1px solid $quinary-content;
                border-radius: 4px;
                align-self: stretch;
                display: flex;
                align-items: center;
                flex-direction: column;

                .mx_VerificationQRCode {
                    height: 144px;
                    width: 144px;
                    image-rendering: pixelated;
                    border-radius: 0;
                }
            }

            .mx_AppDownloadDialog_info {
                font-size: $font-12px;
                color: $tertiary-content;
            }

            .mx_AppDownloadDialog_links {
                display: flex;
                flex-direction: row;
                gap: $spacing-8;

                .mx_AccessibleButton {
                    svg {
                        height: 40px;
                    }
                }
            }
        }
    }

    .mx_AppDownloadDialog_legal {
        p {
            margin: 0;
            font-size: $font-12px;
            color: $tertiary-content;
        }
    }
}
