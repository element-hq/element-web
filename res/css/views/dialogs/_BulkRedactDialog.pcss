/*
Copyright 2024 New Vector Ltd.
Copyright 2021 Robin Townsend <robin@robin.town>

SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only
Please see LICENSE files in the repository root for full details.
*/

.mx_BulkRedactDialog {
    .mx_Checkbox,
    .mx_BulkRedactDialog_checkboxMicrocopy {
        line-height: $font-20px;
    }

    .mx_BulkRedactDialog_checkboxMicrocopy {
        margin-left: 26px;
        color: $secondary-content;
    }
}
