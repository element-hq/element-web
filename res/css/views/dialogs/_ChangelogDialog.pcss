/*
Copyright 2024 New Vector Ltd.
Copyright 2016 Aviral Dasgupta

SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only
Please see LICENSE files in the repository root for full details.
*/

.mx_ChangelogDialog_content {
    max-height: 300px;
    overflow: auto;
}

.mx_ChangelogDialog_li {
    padding: 0.2em;
}
