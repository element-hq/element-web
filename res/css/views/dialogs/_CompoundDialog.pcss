/*
Copyright 2021 The Matrix.org Foundation C.I.C.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

/* -------------------------------------------------------------------------------- */
/* DEV NOTE: This stylesheet covers dialogs listed by the compound, including */
/* over multiple React components. The actual inner contents of the dialog should */
/* be in their respective stylesheets. */
/* -------------------------------------------------------------------------------- */

/* Override legacy/default styles for dialogs */
.mx_Dialog_wrapper.mx_CompoundDialog .mx_Dialog {
    padding: 0; /* we'll manage it ourselves */
    color: $primary-content;
}

.mx_CompoundDialog {
    .mx_Dialog {
        display: flex;
        flex-direction: column;
    }

    .mx_CompoundDialog_header {
        padding: 32px 32px 16px 32px;

        h1 {
            display: inline-block;
            font-weight: var(--cpd-font-weight-semibold);
            font-size: $font-24px;
            margin: 0; /* managed by header class */
        }
    }

    .mx_CompoundDialog_cancelButton {
        @mixin customisedCancelButton;
        /* Align with corner radius of dialog */
        position: absolute;
        top: var(--cpd-space-4x);
        right: var(--cpd-space-4x);
    }

    .mx_CompoundDialog_form {
        display: flex;
        flex-direction: column;
        min-height: 0;
        flex: 1;
    }

    .mx_CompoundDialog_content {
        overflow: auto;
        padding: 8px 32px;
        flex: 1;
    }

    .mx_CompoundDialog_footer {
        padding: 20px 32px;
        text-align: right;

        .mx_AccessibleButton {
            margin-left: 24px;
        }
    }
}

.mx_ScrollableBaseDialog {
    display: flex;
    flex-direction: column;

    width: 544px; /* fixed */
    height: 516px; /* fixed */
    max-width: 100%;
    min-height: 0;
    max-height: 80%;

    .mx_CompoundDialog_footer {
        box-shadow: 0px -4px 4px rgba(0, 0, 0, 0.05); /* hardcoded colour for both themes */
        z-index: 1; /* needed to make footer & shadow appear above dialog content */
    }
}
