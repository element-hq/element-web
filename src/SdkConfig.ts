/*
Copyright 2024 New Vector Ltd.
Copyright 2019-2022 The Matrix.org Foundation C.I.C.
Copyright 2016 OpenMarket Ltd

SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only
Please see LICENSE files in the repository root for full details.
*/

import { Optional } from "matrix-events-sdk";
import { mergeWith } from "lodash";

import { SnakedObject } from "./utils/SnakedObject";
import { IConfigOptions, ISsoRedirectOptions } from "./IConfigOptions";
import { isObject, objectClone } from "./utils/objects";
import { DeepReadonly, Defaultize } from "./@types/common";

// see element-web config.md for docs, or the IConfigOptions interface for dev docs
export const DEFAULTS: DeepReadonly<IConfigOptions> = {
    brand: "Element",
    help_url: "https://element.io/help",
    help_encryption_url: "https://element.io/help#encryption",
    integrations_ui_url: "https://scalar.vector.im/",
    integrations_rest_url: "https://scalar.vector.im/api",
    uisi_autorageshake_app: "element-auto-uisi",
    show_labs_settings: false,
    force_verification: false,

    jitsi: {
        preferred_domain: "meet.element.io",
    },
    element_call: {
        url: "https://call.element.io",
        use_exclusively: false,
        participant_limit: 8,
        brand: "Element Call",
    },

    // @ts-ignore - we deliberately use the camelCase version here so we trigger
    // the fallback behaviour. If we used the snake_case version then we'd break
    // everyone's config which has the camelCase property because our default would
    // be preferred over their config.
    desktopBuilds: {
        available: true,
        // eslint-disable-next-line @typescript-eslint/no-require-imports
        logo: require("../res/img/element-desktop-logo.svg").default,
        url: "https://element.io/get-started",
    },
    voice_broadcast: {
        chunk_length: 2 * 60, // two minutes
        max_length: 4 * 60 * 60, // four hours
    },

    feedback: {
        existing_issues_url:
            "https://github.com/vector-im/element-web/issues?q=is%3Aopen+is%3Aissue+sort%3Areactions-%2B1-desc",
        new_issue_url: "https://github.com/vector-im/element-web/issues/new/choose",
    },

    desktop_builds: {
        available: true,
        logo: "vector-icons/1024.png",
        url: "https://element.io/download",
        url_macos: "https://packages.element.io/desktop/install/macos/Element.dmg",
        url_win64: "https://packages.element.io/desktop/install/win32/x64/Element%20Setup.exe",
        url_win32: "https://packages.element.io/desktop/install/win32/ia32/Element%20Setup.exe",
        url_linux: "https://element.io/download#linux",
    },
    mobile_builds: {
        ios: "https://apps.apple.com/app/vector/id1083446067",
        android: "https://play.google.com/store/apps/details?id=im.vector.app",
        fdroid: "https://f-droid.org/repository/browse/?fdid=im.vector.app",
    },
};

export type ConfigOptions = Defaultize<IConfigOptions, typeof DEFAULTS>;

function mergeConfig(
    config: DeepReadonly<IConfigOptions>,
    changes: DeepReadonly<Partial<IConfigOptions>>,
): DeepReadonly<IConfigOptions> {
    // return { ...config, ...changes };
    return mergeWith(objectClone(config), changes, (objValue, srcValue) => {
        // Don't merge arrays, prefer values from newer object
        if (Array.isArray(objValue)) {
            return srcValue;
        }

        // Don't allow objects to get nulled out, this will break our types
        if (isObject(objValue) && !isObject(srcValue)) {
            return objValue;
        }
    });
}

type ObjectType<K extends keyof IConfigOptions> = IConfigOptions[K] extends object
    ? SnakedObject<NonNullable<IConfigOptions[K]>>
    : Optional<SnakedObject<NonNullable<IConfigOptions[K]>>>;

export default class SdkConfig {
    private static instance: DeepReadonly<IConfigOptions>;
    private static fallback: SnakedObject<DeepReadonly<IConfigOptions>>;

    private static setInstance(i: DeepReadonly<IConfigOptions>): void {
        SdkConfig.instance = i;
        SdkConfig.fallback = new SnakedObject(i);

        // For debugging purposes
        window.mxReactSdkConfig = i;
    }

    public static get(): IConfigOptions;
    public static get<K extends keyof IConfigOptions>(key: K, altCaseName?: string): IConfigOptions[K];
    public static get<K extends keyof IConfigOptions = never>(
        key?: K,
        altCaseName?: string,
    ): DeepReadonly<IConfigOptions> | DeepReadonly<IConfigOptions>[K] {
        if (key === undefined) {
            // safe to cast as a fallback - we want to break the runtime contract in this case
            return SdkConfig.instance || <IConfigOptions>{};
        }
        return SdkConfig.fallback.get(key, altCaseName);
    }

    public static getObject<K extends keyof IConfigOptions>(key: K, altCaseName?: string): ObjectType<K> {
        const val = SdkConfig.get(key, altCaseName);
        if (isObject(val)) {
            return new SnakedObject(val);
        }

        // return the same type for sensitive callers (some want `undefined` specifically)
        return (val === undefined ? undefined : null) as ObjectType<K>;
    }

    public static put(cfg: DeepReadonly<ConfigOptions>): void {
        SdkConfig.setInstance(mergeConfig(DEFAULTS, cfg));
    }

    /**
     * Resets the config.
     */
    public static reset(): void {
        SdkConfig.setInstance(mergeConfig(DEFAULTS, {})); // safe to cast - defaults will be applied
    }

    public static add(cfg: Partial<ConfigOptions>): void {
        SdkConfig.put(mergeConfig(SdkConfig.get(), cfg));
    }
}

export function parseSsoRedirectOptions(config: IConfigOptions): ISsoRedirectOptions {
    // Ignore deprecated options if the config is using new ones
    if (config.sso_redirect_options) return config.sso_redirect_options;

    // We can cheat here because the default is false anyways
    if (config.sso_immediate_redirect) return { immediate: true };

    // Default: do nothing
    return {};
}
