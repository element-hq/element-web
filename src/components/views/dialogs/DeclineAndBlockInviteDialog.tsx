/*
Copyright 2025 New Vector Ltd.

SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only OR LicenseRef-Element-Commercial
Please see LICENSE files in the repository root for full details.
*/

import React, { type ChangeEventHandler, useCallback, useState } from "react";
import { Field, Label, Root } from "@vector-im/compound-web";

import { _t } from "../../../languageHandler";
import BaseDialog from "./BaseDialog";
import DialogButtons from "../elements/DialogButtons";
import LabelledToggleSwitch from "../elements/LabelledToggleSwitch";

interface IProps {
    onFinished: (shouldReject: boolean, ignoreUser: boolean, reportRoom: false | string) => void;
    roomName: string;
}

export const DeclineAndBlockInviteDialog: React.FunctionComponent<IProps> = ({ onFinished, roomName }) => {
    const [shouldReport, setShouldReport] = useState<boolean>(false);
    const [ignoreUser, setIgnoreUser] = useState<boolean>(false);

    const [reportReason, setReportReason] = useState<string>("");
    const reportReasonChanged = useCallback<ChangeEventHandler<HTMLTextAreaElement>>(
        (e) => setReportReason(e.target.value),
        [setReportReason],
    );

    const onCancel = useCallback(() => onFinished(false, false, false), [onFinished]);
    const onOk = useCallback(
        () => onFinished(true, ignoreUser, shouldReport ? reportReason : false),
        [onFinished, ignoreUser, shouldReport, reportReason],
    );

    return (
        <BaseDialog
            className="mx_DeclineAndBlockInviteDialog"
            onFinished={onCancel}
            title={_t("decline_invitation_dialog|title")}
            contentId="mx_Dialog_content"
        >
            <Root>
                <p>{_t("decline_invitation_dialog|confirm", { roomName })}</p>
                <LabelledToggleSwitch
                    label={_t("report_content|ignore_user")}
                    onChange={setIgnoreUser}
                    caption={_t("decline_invitation_dialog|ignore_user_help")}
                    value={ignoreUser}
                />
                <LabelledToggleSwitch
                    label={_t("action|report_room")}
                    onChange={setShouldReport}
                    caption={_t("decline_invitation_dialog|report_room_description")}
                    value={shouldReport}
                />
                <Field name="report-reason" aria-disabled={!shouldReport}>
                    <Label htmlFor="mx_DeclineAndBlockInviteDialog_reason">
                        {_t("room_settings|permissions|ban_reason")}
                    </Label>
                    <textarea
                        id="mx_DeclineAndBlockInviteDialog_reason"
                        className="mx_RoomReportTextArea"
                        placeholder={_t("decline_invitation_dialog|reason_description")}
                        rows={5}
                        onChange={reportReasonChanged}
                        value={shouldReport ? reportReason : ""}
                        disabled={!shouldReport}
                    />
                </Field>
                <DialogButtons
                    primaryButton={_t("action|decline_invite")}
                    primaryButtonClass="danger"
                    cancelButton={_t("action|cancel")}
                    onPrimaryButtonClick={onOk}
                    onCancel={onCancel}
                />
            </Root>
        </BaseDialog>
    );
};
