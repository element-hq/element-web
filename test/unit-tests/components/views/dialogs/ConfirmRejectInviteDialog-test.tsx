/*
Copyright 2025 New Vector Ltd.

SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only OR LicenseRef-Element-Commercial
Please see LICENSE files in the repository root for full details.
*/

import { render } from "jest-matrix-react";
import userEvent from "@testing-library/user-event";
import React from "react";

import SdkConfig from "../../../../../src/SdkConfig";
import { DeclineAndBlockInviteDialog } from "../../../../../src/components/views/dialogs/DeclineAndBlockInviteDialog";

describe("ConfirmRejectInviteDialog", () => {
    const onFinished: jest.Mock<any, any> = jest.fn();

    const MY_ROOM_NAME = "foo";

    beforeEach(() => {
        jest.resetAllMocks();
    });

    afterEach(() => {
        SdkConfig.reset();
    });

    it("can close the dialog", async () => {
        const { getByTestId } = render(<DeclineAndBlockInviteDialog onFinished={onFinished} roomName={MY_ROOM_NAME} />);
        await userEvent.click(getByTestId("dialog-cancel-button"));
        expect(onFinished).toHaveBeenCalledWith(false, false, false);
    });

    it("can reject with options selected", async () => {
        const { container, getByLabelText, getByRole } = render(
            <DeclineAndBlockInviteDialog onFinished={onFinished} roomName={MY_ROOM_NAME} />,
        );
        await userEvent.click(getByRole("switch", { name: "Ignore user" }));
        await userEvent.click(getByRole("switch", { name: "Report room" }));
        await userEvent.type(getByLabelText("Reason"), "I want to report this room");
        expect(container).toMatchSnapshot();
        await userEvent.click(getByRole("button", { name: "Decline invite" }));
        expect(onFinished).toHaveBeenCalledWith(true, true, "I want to report this room");
    });
    it("can reject without a reason", async () => {
        const { getByRole } = render(<DeclineAndBlockInviteDialog onFinished={onFinished} roomName={MY_ROOM_NAME} />);
        await userEvent.click(getByRole("switch", { name: "Ignore user" }));
        await userEvent.click(getByRole("switch", { name: "Report room" }));
        await userEvent.click(getByRole("button", { name: "Decline invite" }));
        expect(onFinished).toHaveBeenCalledWith(true, true, "");
    });
});
