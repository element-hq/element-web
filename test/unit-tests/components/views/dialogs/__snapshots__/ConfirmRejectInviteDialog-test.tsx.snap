// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmRejectInviteDialog can reject with options selected 1`] = `
<div>
  <div
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
  <div
    aria-describedby="mx_Dialog_content"
    aria-labelledby="mx_BaseDialog_title"
    class="mx_DeclineAndBlockInviteDialog mx_Dialog_fixedWidth"
    data-focus-lock-disabled="false"
    role="dialog"
  >
    <div
      class="mx_Dialog_header"
    >
      <h1
        class="mx_Heading_h3 mx_Dialog_title"
        id="mx_BaseDialog_title"
      >
        Decline invitation
      </h1>
    </div>
    <form
      class="_root_19upo_16"
    >
      <p>
        Are you sure you want to decline the invitation to join "foo"?
      </p>
      <div
        class="mx_SettingsFlag"
      >
        <span
          class="mx_SettingsFlag_label"
        >
          <div
            id="mx_LabelledToggleSwitch_:r7:"
          >
            Ignore user
          </div>
          <span
            class="mx_Caption"
            id="mx_LabelledToggleSwitch_:r7:_caption"
          >
            You will not see any messages or room invites from this user.
          </span>
        </span>
        <div
          aria-checked="true"
          aria-describedby="mx_LabelledToggleSwitch_:r7:_caption"
          aria-disabled="false"
          aria-labelledby="mx_LabelledToggleSwitch_:r7:"
          class="mx_AccessibleButton mx_ToggleSwitch mx_ToggleSwitch_on mx_ToggleSwitch_enabled"
          role="switch"
          tabindex="0"
        >
          <div
            class="mx_ToggleSwitch_ball"
          />
        </div>
      </div>
      <div
        class="mx_SettingsFlag"
      >
        <span
          class="mx_SettingsFlag_label"
        >
          <div
            id="mx_LabelledToggleSwitch_:r8:"
          >
            Report room
          </div>
          <span
            class="mx_Caption"
            id="mx_LabelledToggleSwitch_:r8:_caption"
          >
            Report this room to your account provider.
          </span>
        </span>
        <div
          aria-checked="true"
          aria-describedby="mx_LabelledToggleSwitch_:r8:_caption"
          aria-disabled="false"
          aria-labelledby="mx_LabelledToggleSwitch_:r8:"
          class="mx_AccessibleButton mx_ToggleSwitch mx_ToggleSwitch_on mx_ToggleSwitch_enabled"
          role="switch"
          tabindex="0"
        >
          <div
            class="mx_ToggleSwitch_ball"
          />
        </div>
      </div>
      <div
        aria-disabled="false"
        class="_field_19upo_26"
      >
        <label
          class="_label_19upo_59"
          for="mx_DeclineAndBlockInviteDialog_reason"
        >
          Reason
        </label>
        <textarea
          class="mx_RoomReportTextArea"
          id="mx_DeclineAndBlockInviteDialog_reason"
          placeholder="Describe the reason for reporting the room."
          rows="5"
        >
          I want to report this room
        </textarea>
      </div>
      <div
        class="mx_Dialog_buttons"
      >
        <span
          class="mx_Dialog_buttons_row"
        >
          <button
            data-testid="dialog-cancel-button"
            type="button"
          >
            Cancel
          </button>
          <button
            class="mx_Dialog_primary danger"
            data-testid="dialog-primary-button"
            type="button"
          >
            Decline invite
          </button>
        </span>
      </div>
    </form>
    <div
      aria-label="Close dialog"
      class="mx_AccessibleButton mx_Dialog_cancelButton"
      role="button"
      tabindex="0"
    />
  </div>
  <div
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
</div>
`;
